testing/unit-test.md

```diff
@@ -0,0 +1,34 @@
+# Unit test
+
+## Introduction
+Unit test are used to test individual part of the application in isolation. They cover a huge variety of cases so this guideline are very generic.
+
+## Dos
+- Test the API of modules, if something is used outside his context should be robustly tested.
+- Test as documentation: looking at a good set of unit test you should be able to understand how to use a function without reading its internal code.
+- Test edge cases, what happen with undefined parameters, or with empty parameters?
+- Test errors, if a function is expecting a specific type of parameter test that the function is correctly raising errors.
```

Comment:
Also if you have a bug in production that you create a fix for, there should be a corresponding unit test.